backend/fileSystem.js

```diff
@@ -145,6 +147,84 @@ class FileSystemBackend {
         const match = await this.getPackageByName(packageName);
         return match ? match.versions[version] : null;
     }
+    async searchPackages(properties) {
+        // Retrieve listing of all packages
+        await this.syncMetadata();
+        const packages = Object.keys(this._metadata).map(key => this._metadata[key]);
+
+        // Begin with all packages
+        let result = packages;
+
+        // Filter by search query
+        if (typeof properties.query === 'string' && properties.query.length > 0) {
+            // If only searching for names, it's possible to more efficiently search the whole package array
+            let searchProperties = packages;
+
+            // If doing an advanced search with additional properties, make a new array of package properties
+            if (properties.property.length !== 1 || properties.property[0] !== 'name') {
+                // Extract searchable properties
+                searchProperties = packages.map((pack) => {
+                    // Get latest
+                    const latest = pack.versions[pack['dist-tags'].latest];
+
+                    // Return selected properties
+                    return {
+                        name: pack.name,
+                        description: latest.description || pack.description,
+                        author: latest._npmUser.name,
+                        readme: latest.readme || pack.readme,
+                        keywords: latest.keywords,
+                    };
+                });
+            }
+
+            // Set options for searching
+            const options = {
```

Comment:
Alright, let's keep it as is. Maybe just add a small comment above the properties object with a brief explanation. I'm fine with that, since most of these apply directly to how Fusejs will process results.